Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -903,16 +979,58 @@ namespace OpenInfraPlatform {
 					carve::geom::vector<3> endPoint = getPointOnCurve<typename IfcEntityTypesT::IfcClothoid>(clothoid, trim2Vec, trimmingPreference);
 
 					//get L, A, R 
-					
+					// A - Clothoid constant.
 					double A = clothoid->ClothoidConstant * this->UnitConvert()->getLengthInMeterFactor();
-				 
-				
-					//Fresnel equastion 
-					for (int i = 1; i < 200; i++)
+
+					// Calculate the length "L" between two ponts.
+					//double L = sqrt(pow(endPoint.x- firstPoint.x)+pow(endPoint.y - firstPoint.y)+ pow(endPoint.z - firstPoint.z));
+					double L = carve::geom::distance(firstPoint, endPoint);
+
+					// Get number of tesselated segments (note default is 100, write new function!)
+					int numSegments = 100;
+					// Create a tesselated vector
+					std::vector < double > ti(numSegments,0);
+					double pointInterval_ti = (endPoint.x - firstPoint.x) / (numSegments - 1);
+					for (int i = 1; i < ti.size(); ++i)
+					{
+						ti.push_back(ti[i - 1] + pointInterval_ti);
+					}
+					
+					//Calculate clothoid points. Fresnel equastion.
+
+					std::vector<carve::geom::vector<3> > clothoidPoints;
+					clothoidPoints.x[0, 0];
+					clothoidPoints.y[0, 0];
+					clothoidPoints.z[0, 0];
+					
+					for (int i = 1; i < numSegments; ++i) 
+					{
+						clothoidPoints.push_back(carve::geom::vector<3>(
+							carve::geom::VECTOR(
+								clothoidPoint.x[i - 1] + A * sqrt(M_PI)*cos(M_PI / 2 * (ti[i - 1] * ti[i - ]))*pointInterval_ti,
+									clothoidPoint.y[i - 1] + A * sqrt(M_PI)*sin(M_PI / 2 * (ti[i - 1] * ti[i - ]))*pointInterval_ti,
+										0)));

```

Comment:
I'm not convinced by this part either :) Shouldn't this be handled by the function `getPointOnCurve`?

Reply:
Resolved 34bd9d8